src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -126,6 +126,30 @@ private static Project CreateProject(ProjectId projectId, Solution solution)
             return projectState == null ? null : GetProject(projectState.Id);
         }
 
+        /// <summary>
+        /// Given a <paramref name="symbol"/> returns the <see cref="ProjectId"/> of the exact <see cref="Project"/> it
+        /// came from.  This operation is not defined for <see cref="ISymbol"/>s produced from other <see
+        /// cref="Solution"/> snapshots.
```

Comment:
Why can't we define what it does in that case?

Reply:
fair point.  i should instead say that it will return null.